src/python/pants/core/goals/lint.py

```diff
@@ -105,40 +92,37 @@ async def lint(
     options: LintOptions,
     union_membership: UnionMembership,
 ) -> Lint:
-    field_set_collection_types = union_membership[LinterFieldSets]
-    field_set_collections: Iterable[LinterFieldSets] = tuple(
-        field_set_collection_type(
-            field_set_collection_type.field_set_type.create(target_with_origin)
+    lint_request_types = union_membership[LintRequest]
+    lint_requests: Iterable[StyleRequest] = tuple(
+        lint_request_type(
+            lint_request_type.field_set_type.create(target_with_origin)
             for target_with_origin in targets_with_origins
-            if field_set_collection_type.field_set_type.is_valid(target_with_origin.target)
+            if lint_request_type.field_set_type.is_valid(target_with_origin.target)
         )
-        for field_set_collection_type in union_membership[LinterFieldSets]
+        for lint_request_type in union_membership[LintRequest]
     )
-    field_set_collections_with_sources: Iterable[FieldSetsWithSources] = await MultiGet(
-        Get[FieldSetsWithSources](FieldSetsWithSourcesRequest(field_set_collection))
-        for field_set_collection in field_set_collections
+    lint_requests_with_sources: Iterable[FieldSetsWithSources] = await MultiGet(
+        Get[FieldSetsWithSources](FieldSetsWithSourcesRequest(lint_request.field_sets))
+        for lint_request in lint_requests
     )
     # NB: We must convert back the generic FieldSetsWithSources objects back into their
-    # corresponding LinterFieldSets, e.g. back to IsortFieldSets, in order for the union rule to
+    # corresponding LintRequest, e.g. back to IsortFieldSets, in order for the union rule to
```

Comment:
This comment can be removed now, I think. It's more obvious that a `FieldSetsWithSources` needs to be converted back into a `LintRequest` given the different shape. Whereas earlier it wasn't obvious because a `FieldSetsWithSources` has the same shape as, say, an `IsortFieldSets`.